### ISSUE TITLE
Cannot run the evaluation

### ISSUE THREAD

**Issue by RibinMathewChalumattu** [chrishokamp/dynamic-transformer-ensembles#3]
Hi Chris,

I tried to run the repo with the evaluation code as described in the README, but I encountered several runtime errors. I assume this is due to the difference in package versions. Can you provide a complete list of packages and versions, which were used for your project?

Best,
Ribin

**Comment by chrishokamp** [comment]
Thanks for reporting this, let me have a look

**Comment by chrishokamp** [comment]
@RibinMTC I updated `requirements.txt` and put quick installation requirements in the README here: https://github.com/chrishokamp/dynamic-transformer-ensembles#setup--installation

Can you see if that works for you? It worked for me in a fresh conda env on my CPU machine (I didn't try on GPU)

**Comment by RibinMathewChalumattu** [comment]
Thanks Chris, I could run the evaluation. For our multi document summarization project we want to use a newer version of the transformers library. However, multiple methods used in _decoding_utils.py_ are not available anymore in the newer versions(like _prepare_scores_for_generation_ (line 452), _modeling_utils.calc_banned_ngram_token_, _modeling_utils.top_k_top_p_filtering_ etc.). Do you have a suggestion on the best way to modify the code to make it compatible with newer versions of the huggingface transformers library?

**Comment by chrishokamp** [comment]
@RibinMTC ok glad you were able to run it :100: :sweat_smile: 
I haven't looked deeply into how the transformers library code has changed since I did this project. The idea for dynamic ensembles was fully formed in my head and I just looked for the easiest way to implement it, my guess is that it shouldn't be too challenging to re-implement with the new APIs but if I were doing that I would probably just start over because it's not a huge amount of code. If you do start a new codebase feel free to tag me on an issue and i'll be happy to have a look.
